viewing_party/party.py

```diff
@@ -0,0 +1,227 @@
+# ************   Wave 1   ************
+def create_movie(title, genre, rating):
+    # return a dictionary if all parameters are Truthy.
+    # return None if any parameter is Falsy
+
+    if title == None or genre == None or rating == None:
+        return None
+
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    """
+    i.e user_data = {"watched": [{"title": "Title A", "genre": "Horror", "rating": 3.5}]}
+    """
+    # an empty list that user has no movies to watch
+    # add movie to "watched" list inside of user_data
+    user_data["watched"].append(movie)
+
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+    # user_data = {"watchlist": [{title: " "...}]
+    #             "watched": [{title: " "...}]}
+
+    for movie in user_data["watchlist"]:
+        # {title: " "...}
+        if movie["title"] == title:
+            #" "
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+
+    return user_data
+
+
+# ************   Wave 2   ************
+def get_watched_avg_rating(user_data):
+    # user_data = {"watchlist": [{title: " "...}]
+    #             "watched": [{title: " "...}, {"title: " ", "rating": ""}]}
+
+    # return rating 0.0 if rating is empty
+    if len(user_data["watched"]) == 0:
+        return 0.0
```

Comment:
Nice 👍 ! In most cases, it's good to check for a failed case early in your function. This is called a `guard clause` and it's a common pattern found in c, go, and other languages.